impeller/renderer/backend/gles/texture_gles.cc

```diff
@@ -294,6 +297,18 @@ bool TextureGLES::OnSetContents(std::shared_ptr<const fml::Mapping> mapping,
                     data->type,             // type
                     tex_data                // data
       );
+    } else {
+      gl.TexSubImage2D(texture_target,         // target
```

Comment:
If memory serves, the origin is bottom-left to `+y` going up. Don't you need to adjust the coordinates accordingly. I see a test but haven't run the expectations. Just double checking it's not shifted down more than you expect.

Reply:
We should clamp the values to be within the known texture size. Otherwise, the `GL_INVALID_VALUE` be generated and those are a bear to track down.

Reply:
I was actually hoping the test I ran would generate a golden, but that doesn't appear to be the case. At any rate, I haven't verified if this is correct for GLES but based on your explanation here it sounds like it is not.

Reply:
From investigation, updating the coordinates to use the bottom left offset causes INVALID_PARAMETER errors. The documentation claims:

```
xoffset
Specifies a texel offset in the x direction within the texture array.

yoffset
Specifies a texel offset in the y direction within the texture array.
```
So I think the "texel offset" terminology may be referring to the texture data being treated as a 0 indexed array, which would use the top left corner?